### DISCUSSION TITLE
How to find package without change  CMAKE_TOOLCHAIN_FILE to vcpkg.cmake?

### DISCUSSION THREAD

**Discussion by Hlongyu** [microsoft/vcpkg#40719] [Q&A] [answered]
I'am try to build a Qt application. I build Qt from source, and write a demo to use Qt, all good.    
When i use vcpkg to build Qt application. It will check qt dependence, which i don't want. Because qt works well with xkbcommon-x11 version 0.1.0.
```
-- Checking for module 'xkbcommon-x11>=0.4.1'
--   Package 'xkbcommon-x11' not found
CMake Error at /home/sansi/hly/vcpkg/scripts/buildsystems/vcpkg.cmake:859 (_find_package):
  Found package configuration file:

    /usr/local/Qt-6.5.3/lib/cmake/Qt6XcbQpaPrivate/Qt6XcbQpaPrivateConfig.cmake

  but it set Qt6XcbQpaPrivate_FOUND to FALSE so package "Qt6XcbQpaPrivate" is
  considered to be NOT FOUND.  Reason given by package:
```

**Comment by dg0yt** [comment, marked as answer]
> How to find package without change CMAKE_TOOLCHAIN_FILE to vcpkg.cmake?

Unsupported. `vcpkg.cmake` initializes variables (which could be done manually) and it applies wrappers to CMake packages (which isn't worth duplicating manually).

> ~~~
> -- Checking for module 'xkbcommon-x11>=0.4.1'
> --   Package 'xkbcommon-x11' not found

I don't think this is related to the vcpkg toolchain. `vcpkg.cmake` appears in the CMake call stack because it overloads `find_package`, but the failing check is in the Qt package files.
